platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/EventTransactionPool.java

```diff
@@ -77,9 +81,19 @@ public EventTransactionPool() {
      * @param inFreeze
      * 		Indicates if the system is currently in a freeze
      */
-    public EventTransactionPool(final SettingsProvider settings, final BooleanSupplier inFreeze) {
+    public EventTransactionPool(
+            @NonNull final Metrics metrics, final SettingsProvider settings, final BooleanSupplier inFreeze) {
         this.settings = settings;
         this.inFreeze = inFreeze;
+
+        metrics.getOrCreate(
+                new FunctionGauge.Config<>(INFO_CATEGORY, "transEvent", Integer.class, this::getTransEventSize)
```

Comment:
Is there a reason you inline the metric configs in some places, and have a static member variable in others?

Reply:
Yes. The `FunctionGauge` style metrics require the function to be provided to the configuration object, and you can only obtain the function once the object is constructed.

For other types of metrics, it is possible to statically define the metric's configuration.